### PR TITLE
Skip tests on Mono if KOREBUILD_TEST_SKIPMONO is set.

### PR DESCRIPTION
**Pull request by cesarbs**
This change will enable us to run tests on CoreCLR on the CI.

cc @muratg @Eilon @victorhurdugaci

### CODE REVIEW

**Comment by victorhurdugaci** [comment]
Why not use the `FrameworkSkipConditionAttribute`?

File: build/_k-test.shade [hunk 1]
```diff
@@ -58,7 +59,10 @@ projectFile=''
                     testArgs = " -parallel none";
                 }
 
-                Dnx("test" + testArgs, projectFolder);
+                if (!IsLinux || string.IsNullOrEmpty(KOREBUILD_TEST_SKIPMONO))
```
Comment (muratg) [inline build/_k-test.shade]: Sorry, I'm not sure if I understand the logic inside `if`.  When do we execute this command and why the difference between Linux and OS X?

**Comment by cesarbs** [comment]
@victorhurdugaci Because I'd have to put it on every test in every repo. We want to be able to run tests on CoreCLR **only** across the board. The idea is to have two CIs per non-Windows system, one where we run tests on Mono and another where we run tests on CoreCLR.

File: build/_k-test.shade (hunk 1 above)
Comment (cesarbs) [inline build/_k-test.shade]: That's a poorly named variable. It's set to true on OS X.

Comment (victorhurdugaci) [inline build/_k-test.shade]: Check if `1` of `true`, not that that if it's empty or not. Otherwise, setting it to "false" or "foo" will skip the tests too

**Comment by victorhurdugaci** [comment]
Can you wait until I merge the dotnet support (1h or so) and then make the same change in `_dotnet-test` too?

**Comment by cesarbs** [comment]
Sure.

File: build/_k-test.shade (hunk 1 above)
Comment (cesarbs) [inline build/_k-test.shade]: I'll check for specific negative values, `0` and `false`. All else is `true`.

Comment (victorhurdugaci) [inline build/_k-test.shade]: Why?

**Comment by cesarbs** [comment]
Ping.

File: KoreBuild-dotnet/build/_dotnet-test.shade [hunk 2]
```diff
@@ -59,9 +60,18 @@ projectFile=''
...
+                    string.Equals(KOREBUILD_TEST_SKIPMONO, "false"))
+                {
+                    // Dotnet("test" + testArgs, projectFolder);
+                }
             }
-            else if (!IsLinux || !string.IsNullOrEmpty(KOREBUILD_TEST_DNXCORE))
+            else if (!IsLinux ||
+                     (!string.IsNullOrEmpty(KOREBUILD_TEST_DNXCORE) &&
+                      !string.Equals(KOREBUILD_TEST_DNXCORE, "0") &&
+                      !string.Equals(KOREBUILD_TEST_DNXCORE, "false")))
```
Comment (dougbu) [inline KoreBuild-dotnet/build/_dotnet-test.shade]: Conditions are rather odd. Where else do we consider "anything but `0` and `false`" as true? Use `string.IsNullOrEmpty()` to be consistent with existing `KOREBUILD_TEST_DNXCORE` checks. Or switch both checks to treat only `1` and `true` as true, doing part of #326.

**Comment by dougbu** [comment]
:watch:

File: KoreBuild-dotnet/build/_dotnet-test.shade (hunk 2 above)
Comment (cesarbs) [inline KoreBuild-dotnet/build/_dotnet-test.shade]: I just did this in response to @victorhurdugaci saying not to just use string.IsNullOrEmpty...

Comment (cesarbs) [inline KoreBuild-dotnet/build/_dotnet-test.shade]: Also, I don't think it's so odd. Handling `0` and `false` as false and all else as true is somewhat like what some programming languages do.

File: build/_k-test.shade (hunk 1 above)
Comment (Eilon) [inline build/_k-test.shade]: This is getting crazy. Just check for `1` and `true` (lowercase as far as I'm concerned) and ignore everything else.

Comment (cesarbs) [inline build/_k-test.shade]: :+1:

Comment (Eilon) [inline build/_k-test.shade]: BTW the reasoning for this is: options are generally **off** by default, so we want only very particular settings to turn an option **on**. All other values do nothing.

**Comment by cesarbs** [comment]
Checking for `1` or `true` per @Eilon's order :grin:

**Comment by dougbu** [comment]
:shipit:

**Pull request merged**